tools/traci/connection.py

```diff
@@ -50,7 +50,11 @@ def __init__(self, host, port, process, traceFile, traceGetters):
         else:
             self._socket = socket.socket()
         self._socket.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
-        self._socket.connect((host, port))
+        try:
+            self._socket.connect((host, port))
+        except socket.error as e:
+            self._socket.close()
+            raise e
```

Comment:
A more concise option:
```suggestion
        except socket.error:
            self._socket.close()
            raise
```